### PR TITLE
WIP #1105 - Add an_element specific for rational input

### LINKED ISSUE #1105: An element for HPolytope is not type stable
```julia
julia> P
HPolytope{Rational{Int64}}(HalfSpace{Rational{Int64}}[HalfSpace{Rational{Int64}}(Rational{Int64}[1//1, 0//1], 1//1), HalfSpace{Rational
{Int64}}(Rational{Int64}[0//1, 1//1], 1//1), HalfSpace{Rational{Int64}}(Rational{Int64}[-1//1, 0//1], 1//1), HalfSpace{Rational{Int64}}
(Rational{Int64}[0//1, -1//1], 1//1)])

julia> an_element(P)
2-element Array{Float64,1}:
 1.0
 0.0
```

### PR DESCRIPTION
**Pull request by mforets**
Closes #1105.

With respect to the comment [here](https://github.com/JuliaReach/LazySets.jl/issues/1105#issuecomment-462601735), i decided to go for using the method for floating point, then checking membership, if it works good, otherwise use the Chebyshev method (which should always work, i believe).

### CODE REVIEW

File: src/HPolyhedron.jl [hunk 1]
```diff
@@ -454,6 +454,29 @@ function linear_map(M::AbstractMatrix{N},
     return PT(constraints)
 end
 
+# specific an_element function for rational inputs
+function an_element(P::HPolytope{N};
+                    backend=nothing) where {N<:Rational{Int}}
+    # use the floating point method, then convert to rational and check membership
+    p = invoke(an_element, Tuple{HPolytope}, P)
```
Comment (schillic) [inline src/HPolyhedron.jl]: I am quite convinced that this would be an infinite loop.

Comment (mforets) [reply src/HPolyhedron.jl]: Haha i tried it! `Tuple{HPolytope}` uses the `Float64` constructor.

Comment (schillic) [reply src/HPolyhedron.jl]: Okay, then I do not like that exploit :smile:

**Comment by schillic** [comment]
Even if we decide to not use this implementation, we could offer `chebyshev_center` as a new method.

**Comment by schillic** [comment]
Superseded by #1697.

**Pull request closed without merging**